sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClient.java

```diff
@@ -416,7 +421,7 @@ public String getEventHubName() {
         return getSendLink(batch.getPartitionId())
             .flatMap(link -> {
                 if (isTracingEnabled) {
-                    Context userSpanContext = parentContext.get();
+                    Context userSpanContext = sharedContext;
                     Context entityContext = userSpanContext.addData(ENTITY_PATH_KEY, link.getEntityPath());
                     // start send span and store updated context
                     parentContext.set(tracerProvider.startSpan(
```

Comment:
will this span reuse span builder we created before?

Reply:
yes, the sharedContext on [this](https://github.com/Azure/azure-sdk-for-java/pull/6773/files#diff-afcf12d6b6264dbb96ae63baef89082dR402) line, should get us the span builder returned by the `ProcessKind.LINK`.